frontends/election-manager/src/components/export_logs_modal.tsx

```diff
@@ -0,0 +1,283 @@
+              result: 'Logs are not exportable.',
+            }
+          );
+        }
+        setIsLocatingLogFile(false);
+      }
+    }
+    void checkLogFile();
+  }, [currentUserSession, logger]);
+
+  async function exportResults(openFileDialog: boolean) {
+    assert(window.kiosk);
+    assert(currentUserSession); // TODO(auth) should this check for a specific user type
+    setCurrentState(ModalState.SAVING);
+
+    try {
+      const results = await window.kiosk.readFile(
+        `${LOGS_ROOT_LOCATION}/${LOG_NAME}`
+      );
+      let filenameLocation = '';
+      const usbPath = await usbstick.getDevicePath();
+      if (openFileDialog) {
+        const fileWriter = await window.kiosk.saveAs({
+          defaultPath: LOG_NAME,
+        });
+
+        if (!fileWriter) {
+          throw new Error('could not begin download; no file was chosen');
+        }
+
+        await fileWriter.write(results);
+        filenameLocation = fileWriter.filename;
+        await fileWriter.end();
+      } else {
+        assert(typeof usbPath !== 'undefined');
+        const pathToFile = path.join(usbPath, LOG_NAME);
+        assert(window.kiosk);
+        await window.kiosk.writeFile(pathToFile, results);
+        filenameLocation = LOG_NAME;
+      }
+
+      setSavedFilename(filenameLocation);
+      await new Promise((resolve) => setTimeout(resolve, 2000));
+      await logger.log(LogEventId.FileSaved, currentUserSession.type, {
+        disposition: 'success',
+        message: `Successfully saved log file to ${filenameLocation} on the usb drive.`,
+        fileType: 'logs',
+        filename: filenameLocation,
+      });
+      setCurrentState(ModalState.DONE);
+    } catch (error) {
+      setErrorMessage(error.message);
+      await logger.log(LogEventId.FileSaved, currentUserSession.type, {
+        disposition: 'failure',
+        message: `Error saving log file: ${error.message}`,
+        result: 'File not saved, error message shown to user.',
+        fileType: 'logs',
+      });
+      setCurrentState(ModalState.ERROR);
+    }
+  }
+
+  if (currentState === ModalState.ERROR) {
+    return (
+      <Modal
+        content={
+          <Prose>
+            <h1>Failed to Save Logs</h1>
+            <p>Failed to save log file. {errorMessage}</p>
+          </Prose>
+        }
+        onOverlayClick={onClose}
+        actions={<LinkButton onPress={onClose}>Close</LinkButton>}
+      />
+    );
+  }
+
+  if (currentState === ModalState.DONE) {
+    const actions = <LinkButton onPress={onClose}>Close</LinkButton>;
+    return (
+      <Modal
+        content={
+          <Prose>
+            <h1>Logs Saved</h1>
+            <p>
+              Log file successfully saved{' '}
+              {savedFilename !== '' && (
+                <span>
+                  as <strong>{savedFilename}</strong>
+                </span>
+              )}{' '}
+              directly on the inserted USB drive.
```

Comment:
This is only true when you click "Save", not "Save As".

Reply:
Yeah, but kiosk-browser still only has permissions to save on usb drives so its impossible to save it elsewhere (in production), it will not show the folder if you save it in a folder on the usb though so I'm going to remove the word "directly"